src/xGadgetron/cGadgetron/gadgetron_data_containers.cpp

```diff
@@ -549,17 +550,17 @@ AcquisitionsFile::take_over(MRAcquisitionData& ac)
 {
 	AcquisitionsFile& af = (AcquisitionsFile&)ac;
 	acqs_info_ = ac.acquisitions_info();
-	if (index_)
-		delete[] index_;
+	if (index_.size() > 0)
+		index_.resize(0);
 	int* index = ac.index();
 	sorted_ = ac.sorted();
 	if (sorted_ && index) {
```

Comment:
I guess this test on `index` will now fail. I suppose it was possible that it was a 0 pointer before, but now it will never be. I'm not so sure what would happen when calling the new `index()` (i.e. `&index_[0]`) if `index_.size()==0`.

I guess it now has to be
```C++
if (sorted_ && ac.index_.size()>0)
```
or variations on that.

However, @evgueni-ovtchinnikov is the original logic correct? It will set `index_` to empty if `ac` is non-sorted. Is that intentional?

Reply:
no in this case `int* index` is not `index()`, s.t. the if asks if the `ac.index_` object is empty.
I changed the name to `other_index` to distinguish it a bit more and deleted the resize.